lib/cachex/spec.ex

```diff
@@ -470,4 +470,22 @@ defmodule Cachex.Spec do
   @spec wrap(any, atom) :: {atom, any}
   defmacro wrap(value, tag) when is_atom(tag),
     do: quote(do: {unquote(tag), unquote(value)})
+
+  # extracts list of `$callers` from the process dicionary
+  # and adds the current process to the list
+  # the intended use is to save it in a variable
+  # and use it inside spawned code
+  # e.g.
+  #
+  #     callers = get_callers()
+  #     spawn(fn -> put_callers() ... end)
+  #
+  # `$callers` allows mocking libraries like `Mox`
+  # to set mocks in the test process and use them
+  # in other processes that have the test process in `$callers`
+  @doc false
+  def get_callers(), do: [self() | Process.get(:"$callers") || []]
```

Comment:
I know I made a comment before about macros, but I would macro this to keep consistency with the others in here -- and call it `callers()` instead of `get_callers()`. 

I keep going back and forth on this so maybe we just merge as is and I can change after the fact if needed -- after all these are internals nobody has to care about.

Reply:
Er, I think as a follow up I'd combine what you've done in this file a little bit. 

I would have:

```elixir
def callers(),
  do: Process.get(:"$callers", [])

def callers(callers),
  do: Process.put(:"$callers", callers)
```

(as macro, probably).

Reply:
I am not sure if it makes sense to convert functions to macros just for consistency.
If you would like to `Cachex.Spec` for macros only, we could make a separate module just for those two little helpers. E.g. `Cachex.MockHelpers` with `@moduledoc false`.

Personally, I don't like the change to `callers/0` and `callers/1`. If something is named just with a noun, I usually expect, it will be returned and the argument refines the query.

If we go even more detailed with the module name like `Cachex.Callers`, we can shorten the functions to `get` and `put`. `Callers.get()` and `Callers.put(callers)` which is very much in line with basic Elixir data structures like `Map.get(map, key)` and `Map.put(map, key, value)` or even the process dictionary API.